float64 amount
float64 vel
float64 fix_offset
---
float64 end_dist
---
float64 progress